ui.apps/src/main/javascript/minicart/src/components/CreateAccount/createAccount.js

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import React from 'react';
+import { Form } from 'informed';
+import { shape, string, func } from 'prop-types';
+import Field from '../Field';
+import TextInput from '../TextInput';
+import Checkbox from '../Checkbox';
+import Button from '../Button';
+import combine from '../../utils/combineValidators';
+import {
+    validateEmail,
+    isRequired,
+    validatePassword,
+    validateConfirmPassword,
+    hasLengthAtLeast
+} from '../../utils/formValidators';
+import mergeClasses from '../../utils/mergeClasses';
+import defaultClasses from './createAccount.css';
+import useCreateAccount from './useCreateAccount';
+
+const LEAD = 'Check out faster, use multiple addresses, track orders and more by creating an account!';
+
+const CreateAccount = props => {
+    const { showMyAccount } = props;
+    const [{ createAccountError, isSignedIn, isCreatingCustomer }, { createAccount }] = useCreateAccount();
+
+    const handleCreateAccount = formValues => {
+        createAccount(formValues);
+    };
+
+    const errorMessage = createAccountError ? createAccountError : null;
+
+    if (isSignedIn) {
+        showMyAccount();
```

Comment:
I would add a `return` after this line to make sure the evaluation of this component is stopped, since you switch to another view anyway.